framework/src/modules/network/filter_peers.js

```diff
@@ -168,18 +168,22 @@ const getCountByFilter = (peers, filter) => {
  */
 const getConsolidatedPeersList = networkStatus => {
 	const { connectedPeers, newPeers, triedPeers } = networkStatus;
-
-	const uniquerPeersList = [
-		...connectedPeers,
-		...newPeers,
-		...triedPeers,
-	].reduce((uniquePeers, peer) => {
-		const found = uniquePeers.find(
-			findPeer =>
-				findPeer.ip === peer.ipAddress && findPeer.wsPort === peer.wsPort
-		);
-
-		if (!found) {
+	// Assign state 2 to the connected peers
+	const connectedList = connectedPeers.map(peer => {
+		const { ipAddress, options, minVersion, nethash, ...peerWithoutIp } = peer;
+
+		return { ip: ipAddress, ...peerWithoutIp, state: 2 };
+	});
+	// For the peers that are not present in connectedList should be assigned state 0
```

Comment:
Shouldn't it be 1 as in previous versions? https://github.com/LiskHQ/lisk-sdk/blob/40e2e86021ad14661dea78932d638a2a5bb53835/framework/src/modules/chain/logic/peer.js#L197

Reply:
Oh yeah need to change disconnected peer's state to `1` 👍